src/Aspire.Hosting.Azure/AzureOpenAIResource.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure OpenAI resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class AzureOpenAIResource(string name) : Resource(name), IAzureResource, IResourceWithConnectionString
```

Comment:
Are you going to add provisioning as well?

Reply:
I don't think it's possible right now, but wanted to keep an actual resource for the common model.

Reply:
Not possible as in there are no ARM APIs?

Reply:
You can provision the azure open ai account but not deploy models.

I think the best experience here long term would be that you can deploy an LLM with some pre-training but it just doesn't look like model deployment is part of the ARM surface area.

We should talk to the team to hear what their plans are.

Reply:
@mitchdenny they are deployable via Bicep, so should be fine with ARM. [Here is an example](https://github.com/Azure-Samples/azure-search-openai-demo-csharp/blob/main/infra/main.bicep#L333-L358)

Reply:
Thanks @aaronpowell  this is super helpful I was looking for this yesterday and could not find it anywhere (so assumed it wasn' t possible). @sebastienros, given this I think what we want to do here is have an API surface that looks like this:

var aoai = builder.AddAzureOpenAI("aoai").AddDeployment(deploymentName, modelArgs...);

So you would end up with two resource types:

`azure.openai.account.v0`
`azure.openai.deployment.v0`

@aaronpowell do you happen to know if it is possible to send this pre-prompts in via ARM or does that have to be done using the data plane calls?

Reply:
Nope. The model is stateless so things like system prompts need to be sent in every request.

I'm not sure if there is ARM support for doing plugins (BYO data style) yet.

Reply:
Just for curiosity, how do you provision AOAI instance where the designated region is not available for AOAI? For example, I can deploy all resources to `Korea Central` except AOAI. `azd init` and `azd up` assume that all the resources are provisioned in the same region.

When I looked at the code, I thought I would configure the AOAI resource location and the deployment model. Does the `connectionString` include this location and deployment model?